paymentapi/src/test/java/fi/hel/verkkokauppa/payment/api/PaymentAdminControllerUnitTest.java

```diff
@@ -91,4 +145,310 @@ public void savePaymentFilter() throws Exception {
+                    .andExpect(status().is4xxClientError())
+                    .andExpect(status().is(404));
+        });
+
+        CommonApiException cause = (CommonApiException) exception.getCause();
+        assertEquals(CommonApiException.class, cause.getClass());
+        assertEquals("payment-method-not-found", cause.getErrors().getErrors().get(0).getCode());
+        assertEquals("payment method with code [test-payment-code] not found", cause.getErrors().getErrors().get(0).getMessage());
+    }
+
+    /*
+     * It tests the get all payment methods endpoint.
+     */
+    @Test
+    public void whenGetPaymentMethods_thenReturnZeroWithStatus200() throws Exception {
+        Mockito.when(paymentMethodService.getAllPaymentMethods()).thenCallRealMethod();
+        Mockito.when(paymentMethodRepository.findAll()).thenReturn(Collections.emptyList());
+
+        MvcResult response = this.mockMvc.perform(
+                        get("/payment-admin/payment-method/")
+                )
+                .andDo(print())
+                .andExpect(status().is2xxSuccessful())
+                .andExpect(status().is(200))
+                .andReturn();
+        String contentAsString = response.getResponse().getContentAsString();
+        assertEquals(contentAsString, "[]");
+    }
+
+    @Test
+    public void whenGetPaymentMethods_thenReturnMoreThanZeroWithStatus200() throws Exception {
+        PaymentMethodDto paymentMethodDto = createTestPaymentMethodDto(GatewayEnum.OFFLINE);
+        PaymentMethodDto paymentMethodDto2 = createTestPaymentMethodDto(GatewayEnum.ONLINE);
+        paymentMethodDto2.setCode("second-payment-code");
+
+        PaymentMethod paymentMethod = mapper.convertValue(paymentMethodDto, PaymentMethod.class);
+        PaymentMethod paymentMethod2 = mapper.convertValue(paymentMethodDto2, PaymentMethod.class);
+
+        Mockito.when(paymentMethodService.getAllPaymentMethods()).thenCallRealMethod();
+        Mockito.when(paymentMethodRepository.findAll()).thenReturn(Arrays.asList(paymentMethod, paymentMethod2));
+
+        MvcResult response = this.mockMvc.perform(
+                        get("/payment-admin/payment-method/")
+                )
+                .andDo(print())
+                .andExpect(status().is2xxSuccessful())
+                .andExpect(status().is(200))
+                .andReturn();
+
+        List<PaymentMethodDto> responseDtos = Arrays.asList(mapper.readValue(response.getResponse().getContentAsString(), PaymentMethodDto[].class));
+        Assertions.assertEquals(2, responseDtos.size());
+    }
+
+
+
+    private PaymentMethodDto createTestPaymentMethodDto(GatewayEnum gateway) {
+        return new PaymentMethodDto("Test payment method",
+                "test-payment-code",
+                "test-payment-group",
+                "test-payment.jpg",
+                gateway);
+    }
+
+    private static String getFileContents(String filePath) throws URISyntaxException, IOException {
```

Comment:
This is very handy function, you can create Utility class for this in folder, this kind of functions I like a lot! Good job! `src/test/java/fi/hel/verkkokauppa/payment/utils`

FileReader::getContents("")